modules/passkey/contracts/verifiers/IP256Verifier.sol

```diff
@@ -0,0 +1,121 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+/* solhint-disable payable-fallback */
+pragma solidity ^0.8.0;
+
+/**
+ * @title P-256 Elliptic Curve Verifier.
+ * @dev P-256 verifier contract that follows the EIP-7212 EC verify precompile interface. For more
+ * details, refer to the EIP-7212 specification: <https://eips.ethereum.org/EIPS/eip-7212>
+ * @custom:security-contact bounty@safe.global
+ */
+interface IP256Verifier {
+    /**
+     * @notice  A fallback function that takes the following input format and returns a result
+     * indicating whether the signature is valid or not:
+     * - `input[  0: 32]`: message
+     * - `input[ 32: 64]`: signature r
+     * - `input[ 64: 96]`: signature s
+     * - `input[ 96:128]`: public key x
+     * - `input[128:160]`: public key y
+     *
+     * The output is a Solidity ABI encoded boolean value indicating whether or not the signature is
+     * valid. Specifically, it returns 32 bytes with a value of `0x00..00` or `0x00..01` for an
+     * invalid or valid signature respectively.
+     *
+     * Note that this function does not follow the Solidity ABI format (in particular, it does not
+     * have a 4-byte selector), which is why it requires a fallback function and not regular
+     * Solidity function. Additionally, it has `view` function semantics, and is expected to be
+     * called with `STATICCALL` opcode.
+     *
+     * @param input The encoded input parameters.
+     * @return output The encoded signature verification result.
+     */
+    fallback(bytes calldata input) external returns (bytes memory output);
+}
+
+/**
+ * @title P-256 Elliptic Curve Verifier.
+ * @dev P-256 verifier contract that follows the EIP-7212 EC verify precompile interface. For more
+ * details, refer to the EIP-7212 specification: <https://eips.ethereum.org/EIPS/eip-7212>
+ */
+library P256VerifierLib {
```

Comment:
Question: in https://github.com/safe-global/safe-modules/pull/223, I made the verifier wrapper contract an external contract instead of a library to avoid repeating this code in every signer contract and deployed a singleton rather. Have we decided to switch the approach with this PR?

Reply:
While deploying this code as a singleton still could be achieved by deploying a WebAuthnVerifier singleton contract, I think standalone `P256Signers` that do not use webauthn can benefit from it gas-wise

Reply:
Or I'm having a brainfart that doesn't affect anything significantly..

Reply:
> While deploying this code as a singleton still could be achieved by deploying a WebAuthnVerifier singleton contract, I think standalone P256 signers that do not use webauthn can benefit from it gas-wise

The problem is that the P-256 precompile won't have this and need an intermediary contract.

Also, I think this ends up working exactly like #223, as `WebAuthnVerifier` inherits from `P256Wrapper` which calls `P256Verifier`. Here, a potential `WebAuthnVerifier` will import and call `P256VerifierLib` which calls `P256Verifier`. In both cases, the `WebAuthnVerifier` is the one that contains the code for checking malleability and will be a singleton in both cases.